src/app/reducers/vm/redux/vm-creation.effects.ts

```diff
@@ -247,58 +248,63 @@ export class VirtualMachineCreationEffects {
-              })
-              .map((vmWithTags) => {
-                if (action.payload.doStartVm) {
-                  vmWithTags.state = VmState.Running;
-                }
-                return new vmActions.DeploymentRequestSuccess(vmWithTags);
-              })
-              .catch((error) => Observable.of(new vmActions.DeploymentRequestError(error)));
-          })
-          .catch((error) => Observable.of(new vmActions.DeploymentRequestError(error))));
-    });
-
-  @Effect({ dispatch: false })
+        .pipe(
+          switchMap(() => this.doCreateSecurityGroup(action.payload)
+            .pipe(
+              switchMap((securityGroups) => {
+              if (action.payload.securityGroupData.mode === VmCreationSecurityGroupMode.Builder) {
+                this.store.dispatch(new securityGroupActions.CreateSecurityGroupsSuccess(securityGroups));
+              }
+              this.store.dispatch(new vmActions.DeploymentChangeStatus({
+                stage: VmDeploymentStage.SG_GROUP_CREATION_FINISHED
+              }));
+
+              this.handleDeploymentMessages({stage: VmDeploymentStage.VM_CREATION_IN_PROGRESS});
+              const params = this.getVmCreationParams(action.payload, keyboard, securityGroups);
+              let deployResponse;
+
+              return this.vmService.deploy(params)
+                .pipe(
+                  switchMap(response => {
+                    deployResponse = response;
+                    return this.vmService.get(deployResponse.id);
+                  }),
+                  switchMap(vm => {
+                    const temporaryVm = vm;
+
+                    if (action.payload.instanceGroup && action.payload.instanceGroup.name) {
+                      temporaryVm.instanceGroup = action.payload.instanceGroup;
+                    }
+
+                    temporaryVm.state = VmState.Deploying;
+                    this.handleDeploymentMessages({stage: VmDeploymentStage.TEMP_VM});
+
+                    this.store.dispatch(new UserTagsActions.IncrementLastVMId());
+                    return this.vmService.registerVmJob(deployResponse);
+                  }),
+                  switchMap((deployedVm: VirtualMachine) => {
+                    this.handleDeploymentMessages({stage: VmDeploymentStage.VM_DEPLOYED});
+
+                    return this.doCreateInstanceGroup(deployedVm, action.payload)
+                      .switchMap((virtualMachine) => this.doCopyTags(virtualMachine, action.payload));
+                  }),
+                  map((vmWithTags) => {
+                    if (action.payload.doStartVm) {
+                      vmWithTags.state = VmState.Running;
+                    }
+                    return new vmActions.DeploymentRequestSuccess(vmWithTags);
+                  }),
+                  catchError((error) => Observable.of(new vmActions.DeploymentRequestError(error))));
```

Comment:
Use `of` instead of `Observable.of` because it will be deprecated in new rxjs version.
Change it in both `catchError`